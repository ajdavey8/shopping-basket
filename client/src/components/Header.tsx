import React from 'react';
import styled from 'styled-components';

const Container = styled.header`
margin: 16px 40px;
`

interface Header {
  heading: string
}

const Header = ({ heading }: Header) => (
  <Container>
    <h1>{heading}</h1>
  </Container>
)

export default Header;