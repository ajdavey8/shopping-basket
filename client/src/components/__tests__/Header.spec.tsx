import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import Header from '../Header';

const setup = () => {
  const utils = render(
    <Header
      heading='New Title'
    />
  )

  return {
    ...utils
  }
}

describe('<Header>', () => {
  it('renders with the correct header', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})