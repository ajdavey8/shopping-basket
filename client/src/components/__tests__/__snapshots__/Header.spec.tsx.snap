// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders with the correct header 1`] = `[Function]`;
